extension/nine-patch/src/commonMain/kotlin/com/seiko/imageloader/util/NinePatchPainter.kt

```diff
@@ -0,0 +1,208 @@
+package com.seiko.imageloader.util
+
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.FilterQuality
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.toSize
+import com.seiko.imageloader.model.NinePatchCenterRect
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+internal class NinePatchPainter(
+    private val image: ImageBitmap,
+    private val centerSlice: NinePatchCenterRect,
+    private val scale: Float = 1f,
+    private val filterQuality: FilterQuality = FilterQuality.Medium,
+) : Painter() {
+
+    private val centerWidth = max(1, centerSlice.width)
```

Comment:
The minimum centerWidth and centerHeight constraint can be removed since it is already set in NinePatchCenterRect